### ISSUE TITLE
transcript numbers don't match between OUT.transcript_model_grouped_tpm.tsv and OUT.transcript_models.gtf

### ISSUE THREAD

**Issue by dudududu12138** [ablab/IsoQuant#195]
Hi, I recently used isoquant(version 3.4.1) to detect isoforms from multiple samples. But the results are a little odd. Below are my quaetions:
1. There are 167413 transcripts  in `OUT.transcript_models.gtf` while 166857 in `OUT.transcript_model_tpm.tsv`.
2. I didn't see the <combine.*> files. There are only <*grouped*> files. So what is the result file that report the expression levels of multisamples?
3. Below is all my outputs:
```
aux                          OUT.gene_grouped_counts.tsv  OUT.transcript_counts.tsv          OUT.transcript_model_grouped_counts.tsv  OUT.transcript_model_tpm.tsv
OUT.corrected_reads.bed.gz   OUT.gene_grouped_tpm.tsv     OUT.transcript_grouped_counts.tsv  OUT.transcript_model_grouped_tpm.tsv     OUT.transcript_tpm.tsv
OUT.extended_annotation.gtf  OUT.gene_tpm.tsv             OUT.transcript_grouped_tpm.tsv     OUT.transcript_model_reads.tsv.gz
OUT.gene_counts.tsv          OUT.read_assignments.tsv.gz  OUT.transcript_model_counts.tsv    OUT.transcript_models.gtf
```
4. Below is my codes:
```
samples=`ls ../minimap2/*sorted.bam`

isoquant.py -t 30 \
        --reference $ref \
        --genedb $anno --complete_genedb \
        --bam $samples \
        --data_type pacbio_ccs \
        -o $out/
```
Thank you very much!

**Comment by andrewprzh** [comment]
Dear @dudududu12138 

1. Could you send me the files? This looks unexpected.
2. Currently, if you provide several BAM files, they will treated as distinct samples within the same experiment. Thus, a single output folder with a single transcript_models.gtf files will be created. "Per-file" count/TPM tables are stored in _grouped files. 
Combined tables are now created only if you provide 2 or more experiments via YAML dataset file.

Best
Andrey

**Comment by dudududu12138** [comment]
> Dear @dudududu12138
> 
> 1. Could you send me the files? This looks unexpected.
> 2. Currently, if you provide several BAM files, they will treated as distinct samples within the same experiment. Thus, a single output folder with a single transcript_models.gtf files will be created. "Per-file" count/TPM tables are stored in _grouped files.
>    Combined tables are now created only if you provide 2 or more experiments via YAML dataset file.
> 
> Best Andrey

Sorry for the late reply. Below are the `*counts.tsv` and `*gtf` files. There are more transcripts in the gtf file. And some transcript ids are NA in the gtf file. It is odd.
[OUT.transcript_models.gtf.gz](https://github.com/user-attachments/files/15948838/OUT.transcript_models.gtf.gz)
[OUT.transcript_model_counts.tsv.gz](https://github.com/user-attachments/files/15948839/OUT.transcript_model_counts.tsv.gz)

**Comment by dudududu12138** [comment]
By the way, I checked myself. I extracted an example of a transcript that is reported in the `OUT.transcript_models.gtf` file but not in the `OUT.transcript_model_counts.tsv` file.
Below is the transcript id and read id that support the transcript that extracted from `OUT.transcript_model_reads.tsv.gz`.
![1719198912153](https://github.com/ablab/IsoQuant/assets/125525648/40e8733a-3c0a-4370-90b8-8e019b2d1e3e)
Then I extracted the these reads from `OUT.read_assignments.tsv.gz` and saved them into file(
[example.read.assignment.txt](https://github.com/user-attachments/files/15949356/example.read.assignment.txt).

**Comment by andrewprzh** [comment]
@dudududu12138 

Yes, this is really odd.
Could you also send me the full log file?

**Comment by andrewprzh** [comment]
@dudududu12138 

Here is what I found so far. Output GTF and supporting reads are likely correct. However, expression estimation algorithm is independent from transcript discovery and for some reason assigns 0 to some of the transcripts.
Transcripts with 0 counts/TPMs are not printed to `transcript_model_counts.tsv`. Hence the difference.

I will now check why expression estimation algorithm assigns 0 to these transcripts despite the fact they are supported by reads.

Best
Andrey

**Comment by dudududu12138** [comment]
> @dudududu12138
> 
> Here is what I found so far. Output GTF and supporting reads are likely correct. However, expression estimation algorithm is independent from transcript discovery and for some reason assigns 0 to some of the transcripts. Transcripts with 0 counts/TPMs are not printed to `transcript_model_counts.tsv`. Hence the difference.
>  Yes, I also found this reason.
> I will now check why expression estimation algorithm assigns 0 to these transcripts despite the fact they are supported by reads.
> Thanks, looking forward to your checking results!

**Comment by dudududu12138** [comment]
> @dudududu12138
> 
> Yes, this is really odd. Could you also send me the full log file?

This is the log file.
[isoquant.log](https://github.com/user-attachments/files/16095557/isoquant.log)

**Comment by dudududu12138** [comment]
> @dudududu12138
> 
> Yes, this is really odd. Could you also send me the full log file?

**Comment by andrewprzh** [comment]
I think I found the problem. IsoQuant outputs a few novel isoforms that are very similar to the reference ones, and that's where read-to-isoform assignment goes wrong and the novel isoform gets 0 counts because all assignments are ambiguous. So, these similar isoforms should not be there in the first place.

I'll make a bug-fix release soon.

**Comment by andrewprzh** [comment]
Will fixed in 3.5.0.

**Comment by andrewprzh** [comment]
Should be fixed in https://github.com/ablab/IsoQuant/releases/tag/v3.5.0
